decode.go

```diff
@@ -307,7 +328,23 @@ func getData(blob *OSMPBF.Blob) ([]byte, error) {
 	}
 }
 
-func decodeOSMHeader(blob *OSMPBF.Blob) error {
+func (dec *Decoder) readOSMHeader() error {
+	var err error
+	dec.headerOnce.Do(func() {
+		blobHeader, blob, err := dec.readFileBlock()
```

Comment:
I believe the `err` here will shadow the outer `err` making `readOSMHeader()` always return `nil`.